kura/org.eclipse.kura.net.admin/src/main/java/org/eclipse/kura/net/admin/monitor/ModemMonitorServiceImpl.java

```diff
@@ -85,891 +85,933 @@
+		if (oldConfig == null || newConfig == null) {
+			return false;
+		}
+		boolean ret = false;
+		ModemConfig oldModemConfig = getModemConfig(oldConfig);
+		ModemConfig newModemConfig = getModemConfig(newConfig);
+		NetConfigIP4 oldNetConfigIP4 = getNetConfigIp4(oldConfig);
+		NetConfigIP4 newNetConfigIP4 = getNetConfigIp4(newConfig);
+		if (oldNetConfigIP4.equals(newNetConfigIP4) && oldModemConfig.equals(newModemConfig)) {
+			ret = true;
+		}
+		logger.info("old diversity = {},  new diversity = {}", oldModemConfig.isDiversityEnabled(),
+				newModemConfig.isDiversityEnabled());
+		if (oldModemConfig.isDiversityEnabled() != newModemConfig.isDiversityEnabled())
+			ret = false;
+		return ret;
+	}
+
+	private ModemConfig getModemConfig(List<NetConfig> netConfigs) {
+		ModemConfig modemConfig = new ModemConfig();
+		for (NetConfig netConfig : netConfigs) {
+			if (netConfig instanceof ModemConfig) {
+				modemConfig = (ModemConfig) netConfig;
+				break;
+			}
+		}
+		return modemConfig;
+	}
+
+	private NetConfigIP4 getNetConfigIp4(List<NetConfig> netConfigs) {
+		NetConfigIP4 netConfigIP4 = new NetConfigIP4(NetInterfaceStatus.netIPv4StatusUnknown, false);
+		for (NetConfig netConfig : netConfigs) {
+			if (netConfig instanceof NetConfigIP4) {
+				netConfigIP4 = (NetConfigIP4) netConfig;
+				break;
+			}
+		}
+		return netConfigIP4;
+	}
+
+	private int getInterfaceNumber(List<NetConfig> netConfigs) {
+		int ifaceNo = -1;
+		if (netConfigs != null) {
+			for (NetConfig netConfig : netConfigs) {
+				if (netConfig instanceof ModemConfig) {
+					ifaceNo = ((ModemConfig) netConfig).getPppNumber();
+					break;
+				}
+			}
+		}
+		return ifaceNo;
+	}
+
+	private int getInterfaceNumber(String ifaceName) {
+		return Integer.parseInt(ifaceName.replaceAll("[^0-9]", ""));
+	}
+
+	private void setInterfaceNumber(String ifaceName, List<NetConfig> netConfigs) {
+		if (netConfigs != null) {
+			for (NetConfig netConfig : netConfigs) {
+				if (netConfig instanceof ModemConfig) {
+					((ModemConfig) netConfig).setPppNumber(getInterfaceNumber(ifaceName));
+					break;
+				}
+			}
+		}
+	}
+
+	long getModemResetTimeoutMsec(String ifaceName, List<NetConfig> netConfigs) {
+		long resetToutMsec = 0L;
+
+		if (ifaceName != null && netConfigs != null) {
+			for (NetConfig netConfig : netConfigs) {
+				if (netConfig instanceof ModemConfig) {
+					resetToutMsec = ((ModemConfig) netConfig).getResetTimeout() * 60000;
+					break;
+				}
+			}
+		}
+		return resetToutMsec;
+	}
+
+	private boolean isGpsEnabledInConfig(List<NetConfig> netConfigs) {
+		boolean isGpsEnabled = false;
+		if (netConfigs != null) {
+			for (NetConfig netConfig : netConfigs) {
+				if (netConfig instanceof ModemConfig) {
+					isGpsEnabled = ((ModemConfig) netConfig).isGpsEnabled();
+					break;
+				}
+			}
+		}
+		return isGpsEnabled;
+	}
+
+	private boolean isDiversityEnabledInConfig(List<NetConfig> netConfigs) {
+		boolean isDiversityEnabled = false;
+		if (netConfigs != null) {
```

Comment:
I don't like this section: too nested. I know that probably it's a rework of something already there, but maybe the first if can be managed differently. Also, lambdas can be used here.